### ISSUE TITLE
Convert scenery-phet string usage to dynamic strings

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/scenery-phet#780]

**Comment by pixelzoom** [comment]
I'll start working on this.

As noted in 10/24/2022 sprint meeting, I'll probably skip `SceneryPhetStrings.a11y`, because some of the a11y code is still .js, and I'm not familiar with how to test it.

**Comment by zepumph** [comment]
I converted all usages of SceneryPhetStrings outside of scenery-phet to be a StringProperty.

**Comment by pixelzoom** [comment]
@zepumph said:
> I converted all usages of SceneryPhetStrings outside of scenery-phet to be a StringProperty.

Just about everywhere this was done, I see this sort of naming problem, e.g. in GFLBVoicingSummaryDescriber.ts:

```
const singleScreenIntroPatternString = SceneryPhetStrings.a11y.voicing.simSection.screenSummary.singleScreenIntroPatternStringProperty;
```

The const is a Property, so it should have a "Property" suffix, e.g. `singleScreenIntroPatternStringProperty`.

I have a fix coming shortly.

**Comment by pixelzoom** [comment]
"Property" suffix added where appropriate in the above commits. @zepumph feel free to take a peek.

**Comment by pixelzoom** [comment]
@zepumph could you please review https://github.com/phetsims/scenery-phet/commit/ed2a10e88cd4d7422e003f9f562654850224b9ba?  My understanding is that `labelContent` supports type `string | TReadOnlyProperty<string>`.  So `labelString` in SpeedMap entries is unnecessary and undesirable.

**Comment by pixelzoom** [comment]
There are still static strings in 2 files:
- [ ] ScreenSummaryNode.ts
- [ ] SliderControlsKeyboardHelpSection.ts

**Comment by pixelzoom** [comment]
Looks like there's a bug in SliderControlsKeyboardHelpSection.ts, line 90:

```typescript
    const keysStringProperty = ArrowKeyIconDisplay.LEFT_RIGHT === options.arrowKeyIconDisplay ? SceneryPhetStrings.a11y.keyboardHelpDialog.slider.leftRightArrowKeys :
                               ArrowKeyIconDisplay.UP_DOWN === options.arrowKeyIconDisplay ? SceneryPhetStrings.a11y.keyboardHelpDialog.slider.upDownArrowKeys :
                               ArrowKeyIconDisplay.BOTH === options.arrowKeyIconDisplay ?
                               new PatternStringProperty( SceneryPhetStrings.a11y.keyboardHelpDialog.slider.orKeysPatternStringProperty, {
                                 leftRight: SceneryPhetStrings.a11y.keyboardHelpDialog.slider.leftRightArrowKeys,
                                 upDown: SceneryPhetStrings.a11y.keyboardHelpDialog.slider.upDownArrowKeys
                               } ) : null;
    assert && assert( keysStringProperty );
    const keyboardHelpDialogDefaultStepsStringProperty = new PatternStringProperty( SceneryPhetStrings.a11y.keyboardHelpDialog.slider.defaultStepsDescriptionPatternStringProperty, {
      verb: options.verbString,
      slider: options.sliderString,
90    keys: keysStringProperty
    } );
```

`keys: keysStringProperty` is going to effectively be `keysStringProperty.toString()`, not `keysStringProperty.value`.   I believe that you need to use DerivedProperty here instead of PatternStringProperty, because there are 2 dependencies.

@zepumph can you please confirm? I can fix, if you don't have time.

**Comment by pixelzoom** [comment]
Similar bug on line 102, masked by the const name:

```
    const shiftKeysString = ArrowKeyIconDisplay.LEFT_RIGHT === options.arrowKeyIconDisplay ? SceneryPhetStrings.a11y.keyboardHelpDialog.slider.shiftLeftRightArrowKeys :
                            ArrowKeyIconDisplay.UP_DOWN === options.arrowKeyIconDisplay ? SceneryPhetStrings.a11y.keyboardHelpDialog.slider.shiftUpDownArrowKeys :
                            ArrowKeyIconDisplay.BOTH === options.arrowKeyIconDisplay ? new PatternStringProperty( SceneryPhetStrings.a11y.keyboardHelpDialog.slider.orKeysPatternStringProperty, {
                              leftRight: SceneryPhetStrings.a11y.keyboardHelpDialog.slider.shiftLeftRightArrowKeys,
                              upDown: SceneryPhetStrings.a11y.keyboardHelpDialog.slider.shiftUpDownArrowKeys
                            } ) : null;
    assert && assert( shiftKeysString );
    const keyboardHelpDialogSmallerStepsStringProperty = new PatternStringProperty( SceneryPhetStrings.a11y.keyboardHelpDialog.slider.smallerStepsDescriptionPatternStringProperty, {
      verb: options.verbString,
102   keys: shiftKeysString
    } );
```

**Comment by pixelzoom** [comment]
On second thought, I handle SliderControlsKeyboardHelpSection.ts, then have @zepumph review.

**Comment by pixelzoom** [comment]
package.json now contains `"supportsInteractiveDescription": true,`, and SceneryPhetStrings.ts is exclusively `LinkableProperty<string>`.  So I think this issue is done.

@zepumph anything you'd like to add?  Feel free to close.

**Comment by zepumph** [comment]
I searched for `StringProperty = new DerivedProperty` and found a case where I felt like PatternStringProperty was simpler and easier. All the other usages seemed right to use DerivedProperty because of extra logic. If we converted them to PatternStringProperty, it would require mapping a value into a new DerivedProperty instance to pass that to the PatternStringProperty which seems annoying and dumb. Ready to close?
